hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/schema/FetchSchemaOperation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl.compact.schema;
+
+import com.hazelcast.internal.serialization.impl.compact.Schema;
+import com.hazelcast.internal.serialization.impl.compact.SchemaService;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.spi.impl.operationservice.Operation;
+
+import java.io.IOException;
+
+public class FetchSchemaOperation extends Operation implements IdentifiedDataSerializable {
```

Comment:
Do we need to guard them since this is Beta. We are sure that we will break this feature and it will not be backward compatible. 

And what do we need to check for RU scenarios when the Operation is a new one. I was thinking it is needed only when an Operation changes and we want to stay compatible. 
Can you point me to a documentation or to an example ?

Reply:
I guess we can always say - we don't support this because it's beta. But implementing various supporting features (RU, XML, YAML and so on) does create a nicer experience. 

What could happen here is - the customer forms a mixed cluster of 4.2 and 5.0 members. The 5.0 members start sending out new operations like `FetchSchemaOperation` to the cluster which of course ends up with deserialisation exceptions on 4.2 members. There's nothing complex about preventing it - usually we just prevent using a feature which requires cluster coordination before the cluster is actually at some cluster version. Like [here](https://github.com/hazelcast/hazelcast/blob/40d555dc1110da9f63fd66f504c83d4382d0384b/hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapProxyImpl.java#L108). So it translates into something like:
```java
if (clusterVersion.isGreaterOrEqual(V5_0)) {
  useCompactFormat();
} else {
  throw new UnsupportedOperationException("Compact format is allowed on cluster version 5.0");
}
```

Reply:
@mmedenjak I have prepared a pr. 
It feels like this is the best we can do. Note that I have checked exact version rather than greaterOrEqual because we can break it on the next version. 
Can you check that pr out ? If it is ok, I will merge it to this pr. 
https://github.com/sancar/hazelcast/pull/11